addons/Slideshow/src/presenter.js

```diff
@@ -1852,5 +1877,142 @@ function AddonSlideshow_create() {
+                } else {
+                    presenter.play();
+                }
+            }
+        };
+
+        var upArrowHandler = function () {
+            if (!presenter.noAudioPlayer && presenter.configuration.audioLoadComplete) {
+                var volume = presenter.configuration.buzzAudio.getVolume();
+                volume += 10;
+                if (volume > 100) {
+                    volume = 100;
+                }
+                presenter.configuration.buzzAudio.setVolume(volume);
+            }
+        };
+
+        var downArrowHandler = function () {
+            if (!presenter.noAudioPlayer && presenter.configuration.audioLoadComplete) {
+                var volume = presenter.configuration.buzzAudio.getVolume();
+                volume -= 10;
+                if (volume < 0) {
+                    volume = 0;
+                }
+                presenter.configuration.buzzAudio.setVolume(volume);
+            }
+        };
+
+        var mapping = {};
+
+        mapping[keys.SPACE] = spaceHandler;
+        mapping[keys.ARROW_LEFT] = leftArrowHandler;
+        mapping[keys.ARROW_UP] = upArrowHandler;
+        mapping[keys.ARROW_RIGHT] = rightArrowHandler;
+        mapping[keys.ARROW_DOWN] = downArrowHandler;
+
+        try {
+            mapping[keycode]();
+        } catch (er) {};
+    };
+
+    presenter.readSlide = function(index, continueAfterTTS) {
+        if (isWCAGOn) {
+            presenter.pause();
+            if (continueAfterTTS) {
+                presenter.speakWithCallback(
+                    [window.TTSUtils.getTextVoiceObject(
+                        presenter.configuration.slides.content[index].audiodescription,
+                        presenter.configuration.lang)],
+                    readSlideCallback);
+            } else {
+                presenter.speak([window.TTSUtils.getTextVoiceObject(
+                        presenter.configuration.slides.content[index].audiodescription,
+                        presenter.configuration.lang)]);
+            }
+        }
+    };
+
+    function readSlideCallback() {
+        if (presenter.isSpeaking) {
+            presenter.isSpeaking = false;
+            presenter.play();
+        }
+    }
+
+    presenter.getTextToSpeechOrNull = function (playerController) {
+        if (playerController) {
+            return playerController.getModule('Text_To_Speech1');
+        }
+
+        return null;
+    };
+
+    presenter.setWCAGStatus = function (isOn) {
+        isWCAGOn = isOn;
+    };
+
+    presenter.speakWithCallback = function(data, callback) {
+        var tts = presenter.getTextToSpeechOrNull(presenter.playerController);
+
+        if (tts && isWCAGOn) {
+            presenter.isSpeaking = true;
+            tts.speakWithCallback(data, callback);
+        }
+    };
+
+    presenter.speak = function(data) {
+        var tts = presenter.getTextToSpeechOrNull(presenter.playerController);
+
+        if (tts && isWCAGOn) {
```

Comment:
Nie wiem czy to bug czy feature, ale audio jest czytane, kiedy slide'y się zmieniają a my już wyszliśmy z addonu - co moze przerwać czytanie innego tekstu.
W tickecie nie jest to doprecyzowane, więc nie traktuję tego jako błąd.